[nnc] Only lower float conv2d's

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56308 [nnc] Support `pow` on CPU
* **#56289 [nnc] Only lower float conv2d's**

While there's no reason to think non-float32 conv2d's *don't* work,
they're only tested in float32 now.  Since that's the most important use case,
I'd rather restrict the dtypes than spend time testing all the weird dtype
combinations that could possibly happen.

Differential Revision: [D27828495](https://our.internmc.facebook.com/intern/diff/D27828495/)

bertmaher committed:
[nnc] Only lower float conv2d's

While there's no reason to think non-float32 conv2d's *don't* work,
they're only tested in float32 now.  Since that's the most important use case,
I'd rather restrict the dtypes than spend time testing all the weird dtype
combinations that could possibly happen.

Differential Revision: [D27828495](https://our.internmc.facebook.com/intern/diff/D27828495/)

[ghstack-poisoned]

navahgar commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -147,6 +147,16 @@ bool conv2dIsSupported(const torch::jit::Node* node) {
   }
 
+  // Inputs should be float32.  Other dtypes need more testing.
+  auto isFloat = [](const torch::jit::Value* v) {
```
I think you missed calling this lambda.

bertmaher committed:
actually use the code on "[nnc] Only lower float conv2d's"


While there's no reason to think non-float32 conv2d's *don't* work,
they're only tested in float32 now.  Since that's the most important use case,
I'd rather restrict the dtypes than spend time testing all the weird dtype
combinations that could possibly happen.

Differential Revision: [D27828495](https://our.internmc.facebook.com/intern/diff/D27828495/)

[ghstack-poisoned]

navahgar reviewed (approved):
LGTM
